### DISCUSSION TITLE
TypeError: AsyncConnectionPool.__init__() got an unexpected keyword argument 'socket_options'

### DISCUSSION THREAD

**Discussion by shughes-uk** [encode/httpx#2955] [Q&A]
We've seen users running into this in the wild

```python
TypeError: AsyncConnectionPool.__init__() got an unexpected keyword argument 'socket_options'
```

The code is just
```python
async with httpx.AsyncClient(http2=True) as client:
```

Sadly I don't have access to the full stack trace but seems like it might be something in 25.1 vs 24.1, perhaps some incompatibility with httpcore?

**Comment by karpetrosyan** [comment]
It seems like your HTTPX version is higher than the maximum httpcore version it can support. 
Just let the HTTPX choose the httpcore version for you, and I think it will solve the problem.
